src/commands/synthetics/__tests__/utils/internal.test.ts

```diff
@@ -216,6 +216,28 @@ describe('utils', () => {
         const overrides = ['defaultStepTimeout=notANumber']
         expect(() => validateAndParseOverrides(overrides)).toThrow('Invalid number value: notANumber')
       })
+
+      it('should suggest correction for invalid case of valid keys', () => {
+        let overrides = ['startURL=blah']
+        expect(() => validateAndParseOverrides(overrides)).toThrow('Invalid key: startURL. Did you mean: startUrl?')
+
+        // resourceUrlSubstitutionRegexe is missing the plural "s"
+        overrides = ['resourceUrlSubstitutionRegexe=blah']
+        expect(() => validateAndParseOverrides(overrides)).toThrow(
+          'Invalid key: resourceUrlSubstitutionRegexe. Did you mean: resourceUrlSubstitutionRegexes?'
+        )
+
+        overrides = ['startUrlSubstitution=blah']
+        expect(() => validateAndParseOverrides(overrides)).toThrow(
+          'Invalid key: startUrlSubstitution. Did you mean: startUrlSubstitutionRegex?'
+        )
+
+        overrides = ['startUrlS=blah']
+        expect(() => validateAndParseOverrides(overrides)).toThrow('Invalid key: startUrlS. Did you mean: startUrl?')
+
+        overrides = ['startUrlSub=blah']
+        expect(() => validateAndParseOverrides(overrides)).toThrow('Invalid key: startUrlSub. Did you mean: startUrl?')
+      })
```

Comment:
Could you add some tests of keys that are not extrapolated? I'm curious about the arbitrary value of 5 for the threshold.

Reply:
The user could end up writing some gibberish, to which I don't think we should suggest anything. 
In my design POV, we should suggest when we are almost sure we can predict the right word.

> Could you add some tests of keys that are not extrapolated?
sure thing!